src/main/java/com/liferay/faces/util/cache/internal/ConcurrentCacheMaxCapacityLRUImpl.java

```diff
@@ -0,0 +1,218 @@
+/**
+ * Copyright (c) 2000-2017 Liferay, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.liferay.faces.util.cache.internal;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.liferay.faces.util.cache.Cache;
+
+
+/**
+ * A simple {@link Cache} which can be accessed/modified concurrently and limits the cache size by removing the least
+ * recently used entry when a new value is added to the full cache. This implementation locks on writes in order to
+ * ensure that the cache cannot grow infinitely. For more details, see {@link
+ * #removeLeastRecentlyUsedCacheValuesIfNecessary(java.lang.Object)}. A simple alternative would be to use {@link
+ * Collections#synchronizedMap(java.util.Map)} on a {@link LinkedHashMap} with accessOrder set to true. However,
```

Comment:
@stiemannkj1 Please change "link LinkedHashMap" to "link java.util.LinkedHashMap" in order to make IDEs and JavaDoc hyperlinks happy.